tensorly/backend/core.py

```diff
@@ -0,0 +1,854 @@
+import inspect
+import importlib
+import os
+import sys
+import threading
+import types
+import warnings
+
+import numpy as np
+import scipy.linalg
+import scipy.sparse.linalg
+
+
+_STATE = threading.local()
+_DEFAULT_BACKEND = None  # changed in `_initialize_default_backend`
+_KNOWN_BACKENDS = ('numpy', 'mxnet', 'pytorch', 'tensorflow', 'cupy')
+_LOADED_BACKENDS = {}
+
+
+def _initialize_default_backend():
```

Comment:
I prefer to isolate global setup code into functions, and then call it once at a clearly defined spot. Could go either way on this though.

Reply:
In general, I agree. However, in that case, given that this only happens once and calls the `set_backend` function, I think it might be clearer to just move that code to the end of the file and not have it in a function -- this is in any case a minor point.